services/src/test/java/org/keycloak/social/github/GitHubIdentityProviderTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.social.github;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+import org.keycloak.broker.oidc.OAuth2IdentityProviderConfig;
+
+/**
+ * Unit test for {@link org.keycloak.social.github.GitHubIdentityProvider}.
+ *
+ * @author Neon Ngo
+ */
+public class GitHubIdentityProviderTest {
```

Comment:
I believe we should have an integration test for this as well as we're not running unit tests anywhere. I know we cannot test against a real enterprise instance (or can we? :)) but we could at least try to input some fake values in the Admin Console and try to login and check the browser really fails to load the GitHub login page. This way we'll be sure Admin Console works properly with this new settings (even though we won't test the API endpoints but it's not possible without actually performing login).

Could you please add some test like that? You can use [SocialLoginTest](https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/SocialLoginTest.java).

Reply:
I'm sorry, I've just realized the SocialLoginTest doesn't use Admin Console but rather creates IdPs using admin REST endpoints directly. We currently don't have any Admin Console tests for social IdPs, so no need to add one.

But in any case, please add some simple test to the SocialLoginTest even though it won't use Admin Console.

Sorry for the confusion.

Reply:
What type of test are you looking for in SocialLoginTest? FYI: this identity provider previously did not have any test case and I added a unit test for it.

Reply:
Since our options are a bit limited here (we don't have the test environment for this, i.e GitHub Enterprise), I'd imagine something like:
1. Configure GitHub IdP with overiden endpoints using REST API (like other tests in the class do).
2. Invoke login using that IdP and check you were redirected to correct (non-existing) URL.

I don't think we can do much more here.

Reply:
@vmuzikar Can you point me to an example of a test (in SocialLoginTest?) that overrides the IdP's REST endpoint?

Reply:
We don't have such test but it's just a matter of adding new config record to the IdP. You can take some inspiration from [here](https://github.com/keycloak/keycloak/blob/4cd5ace80070c970b859fd15cb4ba80cafddf1a5/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/SocialLoginTest.java#L394)